### PR TITLE
WIP: Add sheltered functionality

### PR DESCRIPTION
**Pull request by luisfn**
Simple backend functionality to register people into a shelter. Would need a good soul with FE ability to do the other side

Note: Never worked with TS before, so pretty much based on  other modules

### CODE REVIEW

File: prisma/migrations/20240511204339_/migration.sql [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*
+  Warnings:
+
+  - You are about to drop the column `sheltered_people` on the `shelters` table. All the data in the column will be lost.
+
+*/
+-- AlterTable
+ALTER TABLE "shelters" RENAME COLUMN "sheltered_people" TO "total_sheltered_people";
```
Comment (luisfn) [inline prisma/migrations/20240511204339_/migration.sql:8]: hope this is a safe change

File: src/sheltered-people/ShelteredPeopleSearch.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            contains: this.formProps.search,
+            mode: 'insensitive',
+          },
+        },
+        {
+          name: {
+            contains: this.formProps.search,
+            mode: 'insensitive',
+          },
+        },
```
Comment (luisfn) [inline src/sheltered-people/ShelteredPeopleSearch.ts:26]: we could search by basically any person data

File: src/sheltered-people/sheltered-people.controller.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  @UseGuards(StaffGuard)
+  async store(@Body() body) {
+    try {
+      const data = await this.shelteredPeopleService.store(body);
+      return new ServerResponse(200, 'Successfully created sheltered person', data);
+    } catch (err: any) {
+      this.logger.error(`Failed to create sheltered person: ${err}`);
+      throw new HttpException(err?.code ?? err?.name ?? `${err}`, 400);
+    }
+  }
```
Comment (luisfn) [inline src/sheltered-people/sheltered-people.controller.ts:60]: maybe we should also update the `shelter.totalShelteredPeople` value here

File: prisma/schema.prisma [hunk 4]
```diff
@@ -85,27 +85,46 @@ model Supply {
...
+  prioritySum          Int      @default(value: 0) @map("priority_sum")
+  latitude             Float?
+  longitude            Float?
+  verified             Boolean  @default(value: false)
+  createdAt            String   @map("created_at") @db.VarChar(32)
+  updatedAt            String?  @map("updated_at") @db.VarChar(32)
 
   shelterManagers ShelterManagers[]
   shelterSupplies ShelterSupply[]
+  shelteredPeople ShelteredPeople[]
```
Comment (luisfn) [inline prisma/schema.prisma:105]: Do we need anything else to list only the people of a specific shelter?

File: makefile [hunk 5]
```diff
@@ -0,0 +1,90 @@
+.PHONY: up down build start stop logs ps prisma-generate prisma-migrate prisma-studio init-db shell-api rebuild clean
```
Comment (luisfn) [inline makefile:1]: Hope everybody like makefiles, I think it's much better to handle common commands

**Comment by Kieroth29** [comment]
The idea is really good, but I believe we must be cautious while dealing with sensitive data, and collecting consent forms to comply with LGPD would be even harder. I could help with implementation of data encryption methods and support from lawyers/law representatives might provide a better platform to build this kind of feature, but the discussion needs to mature a bit in my opinion (just pointing I really liked the idea, but thought those were important aspects to be mentioned).

**Comment by luisfn** [comment]
> The idea is really good, but I believe we must be cautious while dealing with sensitive data, and collecting consent forms to comply with LGPD would be even harder. I could help with implementation of data encryption methods and support from lawyers/law representatives might provide a better platform to build this kind of feature, but the discussion needs to mature a bit in my opinion (just pointing I really liked the idea, but thought those were important aspects to be mentioned).

Didn't realised that, you have a very fair point. Let's see what others are thinking about it and we can improve solution from after some discussion

**Pull request closed without merging**